.github/workflows/ci.yml

```
@@ -343,16 +336,73 @@ jobs:
         git clone -b master --single-branch --depth=1 https://github.com/7br/gstest.git ${GS_TEST_DIR}
 
     - name: Basic Test
-      if: steps.nx-filter.outputs.basic == 'true'
+      if: steps.nx-filter.outputs.basic == 'true' || github.ref == 'refs/heads/main'
       env:
         DEPLOYMENT: ${{ matrix.deployment }}
         GS_TEST_DIR: ${{ github.workspace }}/gstest
       run: |
         python3 -m pytest --exitfirst -s -v \
             $(dirname $(python3 -c "import graphscope; print(graphscope.__file__)"))/nx/tests
 
+  networkx-algo-and-generator-test:
+    runs-on: ubuntu-20.04
+    needs: [build-wheels, changes]
+    if: ${{ needs.changes.outputs.networkx == 'true'  || github.ref == 'refs/heads/main' }}
```

push to main run basic test is enough
```suggestion
    if: ${{ needs.changes.outputs.networkx == 'true' }}
```